wgpu-hal/src/metal/device.rs

```diff
@@ -68,18 +68,49 @@ impl super::Device {
         naga_stage: naga::ShaderStage,
     ) -> Result<CompiledShader, crate::PipelineError> {
         let stage_bit = map_naga_stage(naga_stage);
+
+        let module = &stage.module.naga.module;
+        let ep_resources = &layout.per_stage_map[naga_stage];
+
+        let options = naga::back::msl::Options {
+            lang_version: match self.shared.private_caps.msl_version {
+                mtl::MTLLanguageVersion::V1_0 => (1, 0),
+                mtl::MTLLanguageVersion::V1_1 => (1, 1),
+                mtl::MTLLanguageVersion::V1_2 => (1, 2),
+                mtl::MTLLanguageVersion::V2_0 => (2, 0),
+                mtl::MTLLanguageVersion::V2_1 => (2, 1),
+                mtl::MTLLanguageVersion::V2_2 => (2, 2),
+                mtl::MTLLanguageVersion::V2_3 => (2, 3),
+                mtl::MTLLanguageVersion::V2_4 => (2, 4),
+            },
+            inline_samplers: Default::default(),
+            spirv_cross_compatibility: false,
+            fake_missing_bindings: false,
+            per_entry_point_map: naga::back::msl::EntryPointResourceMap::from([(
+                stage.entry_point.to_string(),
+                ep_resources.clone(),
+            )]),
```

Comment:
Yeah, there is a possible improvement we could make so that we skip the other entry points altogether and also don't create a `HashMap` every time by exposing a `write_string_for_ep` fn in naga and using it here instead - but I think it can be another PR.